### ISSUE TITLE
"struct_field" predicate can not be opened for short/unsigned short fields.

### ISSUE THREAD

**Issue by necto** [verifast/verifast#31]
I want to turn my struct predicate into a list of bytes. For that I open every field until I get raw numbers (int, unsigned int, char, short) which I then will convert to byte arrays. But I can not open the predicates for short fields:

```C
struct A {
  int x;
  unsigned short y;
};

void to_verify()
//@ requires true;
//@ ensures true;
{
  struct A a;
  //@ open A_x(&a, _);
  //@ assert A_y(&a, _);
  //@ open A_y(&a, _);
}
```
--> **No such predicate instance.**

It works for `x` - the int field, but does not work for `y` - the short field, even though assert sees the predicate.

**Comment by necto** [comment]
I also can not close it, when converting an array of bytes to the struct.

**Comment by btj** [comment]
Pending me implementing this, see https://groups.google.com/forum/#!topic/verifast/grGwdUnK4BU on how to implement this yourself.

**Comment by btj** [comment]
Fixed by 9fbfea79643fdcaa05849e9fd1e05304885c8d88
